vm/src/format.rs

```diff
@@ -567,32 +582,99 @@ impl FromStr for FormatSpec {
 }
 
 #[derive(Debug, PartialEq)]
-enum FormatPart {
-    AutoSpec(String),
-    IndexSpec(usize, String),
-    KeywordSpec(String, String),
-    Literal(String),
+pub(crate) enum FieldNamePart {
+    Attribute(String),
+    Index(usize),
+    StringIndex(String),
 }
 
-impl FormatPart {
-    fn is_auto(&self) -> bool {
-        match self {
-            FormatPart::AutoSpec(_) => true,
-            _ => false,
+impl FieldNamePart {
+    fn parse_part(
+        chars: &mut impl PeekingNext<Item = char>,
+    ) -> Result<FieldNamePart, FormatParseError> {
+        let ch = chars.next().unwrap();
```

Comment:
I managed to avoid the need for this `unwrap` here by changing the return type to `Result<Option<FieldNamePart>, FormatParseError>`, which also made the use of this function more idiomatic.